lib/DocumentInspectorFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager;
+
+/**
+ * Document inspector factory for the default DocumentInspector.
+ */
+class DocumentInspectorFactory implements DocumentInspectorFactoryInterface
+{
+    /**
+     * @var PathSegmentRegistry
+     */
+    private $pathSegmentRegistry;
+
+    /**
+     * @var DocumentInspector
+     */
+    private $inspector;
+
+    public function __construct(PathSegmentRegistry $pathSegmentRegistry)
+    {
+        $this->pathSegmentRegistry = $pathSegmentRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInspector(DocumentManagerInterface $manager)
+    {
+        if ($this->inspector) {
+            return $this->inspector;
+        }
+
+        $this->inspector = new DocumentInspector(
+            $manager->getRegistry(),
+            $this->pathSegmentRegistry,
+            $manager->getProxyFactory()
+        );
+
+        return $this->inspector;
```

Comment:
From what I see here I can't recognize why a `DocumentInspectorFactory` is necessary. Once the value is cached always the same inspector is returned. Is it necessary to have one of them for each `DocumentManager`? Then they should also be cached separately, otherwise I would remove this class.